http4s/server/src/main/scala/endpoints/http4s/server/BasicAuthentication.scala

```diff
@@ -0,0 +1,34 @@
+package endpoints.http4s.server
+
+import endpoints.algebra.BasicAuthentication.Credentials
+import endpoints.algebra.Documentation
+import org.http4s
+import org.http4s.{BasicCredentials, Status}
+import org.http4s.headers.Authorization
+
+trait BasicAuthentication[F[_]]
+    extends Endpoints[F]
+    with endpoints.algebra.BasicAuthentication {
+
+  private[endpoints] def basicAuthenticationHeader
+    : RequestHeaders[Credentials] =
+    headers =>
+      headers
+        .get(Authorization)
+        .map { authHeader =>
+          authHeader.credentials match {
+            case BasicCredentials(username, password) =>
+              Right(Credentials(username, password))
+            case _ => Left(badRequestResponse)
+          }
+        }
+        .getOrElse(Left(badRequestResponse))
```

Comment:
In case of failure we should return a `401` status code and a header `WWW-Authenticate`.